app/Http/Controllers/Service/CentreController.php

```diff
@@ -34,4 +35,154 @@ public function printRegistrations(Centre $centre)
+        $user = Auth::user();
+
+        // Get now()
+        $now = Carbon::now();
+
+        // Get centres
+        $centres = $user->relevantCentres();
+
+        // get registrations and order them
+        $registrations = Registration::whereIn('centre_id', $centres->pluck('id')->all())
+            ->with(['centre','family.children','family.carers'])
+            ->orderBy('centre_id', 'asc')
+            ->get();
+
+        // set blank rows for laravel-excel
+        $rows = [];
+
+        // Looks like larevel-excel can't auto-generate headers
+        // by collating all the row keys and normalising.
+        // So we have to do it by hand.
+
+        // create base headers
+        $headers = [
+        ];
+
+        // Per registration...
+        foreach ($registrations as $reg) {
+            $row = [
+                "RVID" => $reg->family->rvid,
+                "Centre" => $reg->centre->name,
+                "Primary Carer" => $reg->family->carers->first()->name,
+                "Food Chart Received" => (!is_null($reg->fm_chart_on)) ? $reg->fm_chart_on->format('d/m/Y') : null,
+                "Food Diary Received" => (!is_null($reg->fm_diary_on)) ? $reg->fm_diary_on->format('d/m/Y') : null,
+                "Entitlement" => $reg->family->entitlement,
+            ];
+
+            // Per child dependent things
+            $kids = [];
+            $due_date = null;
+            $eligible = 0;
+            foreach ($reg->family->children as $index => $child) {
+                // make a 'Child X DoB' key
+                $child_index = $index + 1;
+                $status = $child->getStatus();
+                $dob_header = 'Child ' . (string)$child_index . ' DoB';
+
+                // Arrange kids by eligibility
+                switch ($status['eligibility']) {
+                    case 'Pregnancy':
+                        $due_date = $child->dob->format('d/m/Y');
+                        break;
+                    case 'Eligible':
+                        $kids[$dob_header] = $child->dob->format('m/Y');
+                        $eligible += 1;
+                        break;
+                    case "Ineligible":
+                        $kids[$dob_header] = $child->dob->format('m/Y');
+                        break;
+                }
+            }
+            // Add count oif eligible kids
+            $row["Eligible Children"] = $eligible;
+
+            // Add our kids back in
+            $row=array_merge($row, $kids);
+
+            // TODO: What happens for families with two concurrent pregnancies?
+            $row["Due Date"] = $due_date;
+            $row["Leaving Date"] = null;
+
+            // update the headers if necessary
+            if (count($headers) < count($row)) {
+                $headers = array_keys($row);
+            }
+            // stack new row onto the array
+            $rows[] = $row;
+        }
+
+        // en-sparsen the rows with empty fields for unused header.
```

Comment:
Could you fill them all with N/A or null instead? Is it an issue because multiple children are maybe?

This might be a good solution for keeping columns to minimum but seems over complicated?

Reply:
The original doc i used as the inspiration had 3 BUT that's because the old forms had 3. However,  some reports may be for registrations with loads of children. Child DoB columns are therefore strechy for the whole dataset. This does mean that some reports will only have A single DoB column i guess, if all registrations in that report are for single kids families.

Reply:
Just sad. Not confused. :)